src/main/java/net/datafaker/annotations/FakeResolver.java

```diff
@@ -9,26 +9,33 @@
 
 public class FakeResolver<T> {
 
+    private static final JavaObjectTransformer jTransformer = new JavaObjectTransformer();
+
     private final Class<T> clazz;
 
     public FakeResolver(Class<T> clazz) {
         this.clazz = clazz;
     }
 
-    public T generate(Schema<Object, ?> schema) {
+    public T generateFromDefaultSchema() {
         checkFakeAnnotation(clazz);
 
         FakeForSchema fakeForSchemaAnnotation = clazz.getAnnotation(FakeForSchema.class);
-        Schema<Object, ?> useSchema = getSchema(schema, fakeForSchemaAnnotation.value());
+        Schema<Object, ?> useSchema = getSchema(fakeForSchemaAnnotation.value());
 
-        JavaObjectTransformer jTransformer = new JavaObjectTransformer();
         return (T) jTransformer.apply(clazz, useSchema);
     }
 
-    private Schema<Object, ?> getSchema(Schema<Object, ?> schema, String[] pathToSchema) {
-        if (schema != null) {
-            return schema;
-        } else if (pathToSchema.length != 0) {
+    public T generateFromCustomSchema(Schema<Object, ?> schema) {
```

Comment:
why can't we keep same method name `public T generate(Schema<Object, ?> schema)` ?

Reply:
Made one `public T generate(Schema<Object, ?> schema)` method.